### ISSUE TITLE
mtl.how/sim folder management for "MTL all"

### ISSUE THREAD

**Issue by staceypark** [lzim/teampsd#1039]
**1. Add description:** 

Create workflow for managing world creation requests by "MTL for all" teams 

 
**2. Click on the right:**  
i. **Pipelines** - prioritize in a TeamPSD (& Workgroup) pipeline  
ii. **Assignee** - assign to an individual team member producing work for the issue  
iii. **Label**- assign workgroup (lead) responsible for tracking the task  
iv. **Projects** - assign to domain-specific TeamPSD tracker (issues, features, document, manuscript, team)  
v. **Milestones** - select monthly sprint milestone within the workgroup   
vi. **Estimate** - Estimate task in complexity points   
vii. **Epics** - assign to team-wide epic and TeamPSD master plan project  
viii. **Releases** - assign to key team release grouped with other relevant issues  

**3. Once you hit "Submit new issue**, underneath the first comment:   
viiii. **Add dependency** - assess for and add dependencies *blocking* & *blocked by* this issue

**Comment by staceypark** [comment]
@jamesmrollins @lzim 

Short of creating some automated function for this, I think there's two ways we can handle this:

1. From mtl.how/data
![Screen Shot 2020-10-16 at 5 08 37 PM](https://user-images.githubusercontent.com/31089501/96323510-3dd74900-0fd2-11eb-9769-4b067d439e8b.png)

- We could add a third button after "Request New Team Folder" and "Request Team Membership Change" called "Request Sim UI login".
- Ash has been working on the InfoPath backend of these two buttons for #1224, so I'm sure we could get him to add a third button.
- I'm imagining a dialogue asking for their current team name (since they would already have one from the data items) and then a excel sheet they can fill out with the names and email addresses (aka the template we already fill out to create teams on Epicenter).
- Right now, when people use these buttons, email notifications get automatically sent to Team Folder Administrators (Jenn & I) which are the same people who would create these worlds anyways.



2. From mtl.how/sim
![Screen Shot 2020-10-16 at 5 11 29 PM](https://user-images.githubusercontent.com/31089501/96323602-a3c3d080-0fd2-11eb-950c-2c93431d0b47.png)

- In the login box, after the text about which browser to use, we could add a message **Request simulation login by filling out the template here and send it to mtl.help@va.gov.**
- We could link the excel sheet template we use in the backend within that message.


-------
I'm leaning toward option 1 as the least amount of changes to documentation and something that can easily be added to existing functions/resources we already have.

**Comment by jamesmrollins** [comment]
@staceypark I think we should discuss this, so I have a clear understanding of the use case.  On the surface, though, yes, we could put a "request login" line similar to the "Reset Password" line.  It would then provide a popup to gather whatever information you want and automatically send it to wherever you want. 

I think your instinct for higher automation is correct - If they register on the backend, how could we automatically matriculate the registration through the whole system without user or admin intervention?

**Comment by staceypark** [comment]
It would definitely be great if we could have this automate all the way through.
Since we do not have a function for team data file to automatically upload to Epicenter from mtl.how/data, we will need a notification system as well to ping Jenn/I to upload their data in the backend once their worlds are automatically created through the function.

> @staceypark I think we should discuss this, so I have a clear understanding of the use case. On the surface, though, yes, we could put a "request login" line similar to the "Reset Password" line. It would then provide a popup to gather whatever information you want and automatically send it to wherever you want.
> 
> I think your instinct for higher automation is correct - If they register on the backend, how could we automatically matriculate the registration through the whole system without user or admin intervention?

**Comment by lzim** [comment]
Added to **support_workgroup meeting** for Wed. 10/21.

**Comment by jamesmrollins** [comment]
In preparation for support_workgroup meeting Wed 10/21, I have prepared the following comments and questions.  These are also contained within the Agenda for the meeting.

1.  What is overall deadline for this (Wave 1 milestone shows past-due 5 months)
2.  Can we pass information directly outside the VA domain using a Restful API?  @anthonycpichardo, we may need to set up an experiment.
3.  Is there an Epicenter API that will allow for remote registration?  I think possibly yes, since users are self-registering with the MTL Demo site.
4.  Team name (groupID) will likely be the key field that lands the person in the right world.
5.  Are we proposing the trigger to be the registration of the team at the SharePoint site, where they get their team data?
6.  If passing data outside the domain is not possible, then we could create a self-registration feature (as @staceypark proposed above) in the sim ui login page. BUT we have no way to identify the veracity of the user, since we don't have access to active directory.  In other words, someone could easily impersonate a user.  We may need to provide some kind of verification step or code.

**Comment by jamesmrollins** [comment]
Regarding # 1 above:  This is not an immediate high priority (next 30 days), use opportunity capacity. Need to monitor with care, as the demand for user registration management could really sneak up on @lijenn and @staceypark .  @anazariz and @jamesmrollins will seek opportunities to test platform integration.

**Comment by jamesmrollins** [comment]
@staceypark @anazariz @anthonycpichardo  @hirenp-waferwire I have validated the following APIs are available to automate the auto-registration process once a participant list arrives in Epicenter.

1. Group creation – group API
2. Use creation – user API
3. Group membership – membership API
4. World assignment – world API

Next we should investigate a Restful API to deliver the data payload.  I will discuss with my DEV team tonight.

**Comment by staceypark** [comment]
@jamesmrollins Is there an update from the last comment about the Restful API?

**Comment by jamesmrollins** [comment]
@staceypark  - Yes, GitHub has a Restful API we can use.  What's next?  Is this related to @anthonycpichardo 's larger data flows automation processes due in JAN-FEB sometime?

**Comment by staceypark** [comment]
FYI: @lzim @lijenn @anazariz @jamesmrollins 

@anthonycpichardo and I re-read and discussed this issue.
Our understanding is that this needs to be a connection between mtl.how/data <> mtl.how/sim where users input the info through a page at mtl.how/data and the rest of the sim group creation is automatically finished through an API to the sim.

Anthony says this will likely need to be part of the BISL migration (April) in terms of capacity unless we need to prioritize it earlier.
Stacey thinks that we can wait until then unless others disagree. Moving it to backlog for now.

**Comment by lzim** [comment]
@jamesmrollins 
@staceypark 
@lijenn 
@anthonycpichardo 

This needs to be **identified with a Story** in the title and discussed in relation to **Teams/Sharepoint Set Up in 365.**

I've edited for March.

**Comment by lzim** [comment]
This needs to be broken down by **users** and into **weekly point estimates** and aligned with **stories on our roadmap**

**Comment by anthonycpichardo** [comment]
From HQ Workgroup 03/22/2021:

#1039 - "MTL for All" how would they get their data or sim usage enabled? Move to April Epic.

Story 8 will streamline this, but there will still be an admin role to play a) granting permission, b) but could be set up with an individual world.
Stacey was imagining a button on the Sim UI that says, "Request Sim UI login."
Helps them set themselves up somehow.

Icebox: Going across the platforms, doesn't solve a pain point.

TOP PRIORITY 
- Story 8 - reduce MTL admin pain point for team set up.

April to May Epic

Sim UI button - create individual world

April to May Epic

Data UI link - to Sim UI to create individual world.

**Comment by anazariz** [comment]
@staceypark is this issue still valid?

**Comment by jamesmrollins** [comment]
@lijenn this is the original request for the capability you described in the sim_ui_workflow channel.  I think we have a clearer understanding of the need today, then we did two years ago when this request was opened.  I think we should close this request, reference it in the new feature card, and start fresh.  What do you think?  FYI @lzim

**Comment by jamesmrollins** [comment]
Hi @lijenn and @lzim  - The only issue we may face with Facilitator Self-Registration is security.  We may need some way to limit access to only bonafide facilitators.  Currently, we provide those bonafides when we invite them in.   Do we want to provide a qualifying process? Or do we want anyone with a valid VA credential to be able to set up a facilitator account?

**Comment by lzim** [comment]
Picked up with the _MTL 3.7_ release.
